src/Rules/PHPUnit/AssertEqualsIsDiscouragedRule.php

```diff
@@ -55,6 +55,14 @@ public function processNode(Node $node, Scope $scope): array
 			&& ($leftType->isSuperTypeOf($rightType)->yes())
 			&& ($rightType->isSuperTypeOf($leftType)->yes())
 		) {
+			if (strtolower($node->name->name) === 'assertnotequals') {
+				return [
+					RuleErrorBuilder::message(
+						'You should use assertNotSame() instead of assertNotEquals(), because both values are scalars of the same type',
```

Comment:
Would be nicer to just parametrize the existing return with `sprintf` instead of adding a whole new branch where most of the things are the same.